### DISCUSSION TITLE
Toast not showing using useEffect

### DISCUSSION THREAD

**Discussion by luis-viegas** [shadcn-ui/ui#2783] [Q&A]
Hello. I am trying to render a toast after a redirect from a server action, but first, I need to be able to render a toast on page load.

```
export default function Page() {
  const { toast } = useToast();
  useEffect(() => {
    console.log("hello");
    toast({
      title: "Test",
      description: "Test",
    });
  }, [toast]);
```
  
  I can't figure out why this does not work... any help?

Thanks in advance!

**Comment by lucasfelixs** [comment]
Hello @luis-viegas, I don't know if I understand perfectly, maybe it would be good if you created a codepen with the example, but I understand that you want to show a toast before the redirection, if so, you can just invoke the toast function before the redirection, I want say, where you are dispatching the redirect, you can simply invoke the toast beforehand. if I got it wrong, can you give us more details?

**Reply by luis-viegas** [reply]
Does invoking the toast before redirection work? Won't it be invoked in the page before the redirect thus never being shown?

Nevertheless, I have created a stackblitz that illustrates the problem that I think exists: https://stackblitz.com/edit/stackblitz-starters-hzwlsc?file=app%2Ftest%2Fpage.tsx

**Reply by lucasfelixs** [reply]
Well, from what I understand you want to show two toasts, right? The first one to show a loading feedback and the last one after redirection, is that it? If yes, you can create [route events](https://medium.com/@shrsthprios/adding-route-events-in-next-13-for-loading-indicator-during-page-switch-6e4b479886c6) to show the toast while loading the redirect.
